cmd/stakercli/daemon/daemoncommands.go

```diff
@@ -549,3 +551,18 @@ func withdrawableTransactions(ctx *cli.Context) error {
 
 	return nil
 }
+
+func NewStakerServiceJSONRPCClient(remoteAddressWithoutAuth string) (*dc.StakerServiceJSONRPCClient, error) {
+	parsedURL, err := url.Parse(remoteAddressWithoutAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	user, pwd, err := cmd.GetEnvBasicAuth()
+	if err != nil {
+		return nil, err
+	}
+	parsedURL.User = url.UserPassword(user, pwd)
+
+	return dc.NewStakerServiceJSONRPCClient(parsedURL.String())
```

Comment:
We can leave some comment that underneath this functiol will set up basic auth header